packages-next/fields-document/src/relationship-data.tsx

```diff
@@ -41,16 +41,31 @@ export function addRelationshipData(
       }
       return [];
     } else {
+      // Inline
```

Comment:
This isn't right, this case also applies for non-many relationships in component block props.